lib/worker.js

```diff
@@ -0,0 +1,211 @@
+'use strict';
+
+if (!process.features.experimental_workers) {
+  throw new Error('Experimental workers are disabled');
+}
+
+const util = require('util');
+const assert = require('assert');
+const EventEmitter = require('events');
+const WorkerContextBinding = process.binding('WorkerContext');
+const JSONStringify = function(value) {
+  if (value === undefined) value = null;
+  return JSON.stringify(value);
+};
+const JSONParse = JSON.parse;
+const EMPTY_ARRAY = [];
+
+const builtinErrorTypes = new Map([
+  Error, SyntaxError, RangeError, URIError, TypeError, EvalError, ReferenceError
+].map(function(Type) {
+  return [Type.name, Type];
+}));
+
+var Worker = WorkerContextBinding.JsConstructor;
+util.inherits(Worker, EventEmitter);
+
+Worker.prototype._init = function(entryModulePath, options) {
+  if (typeof entryModulePath !== 'string')
+    throw new TypeError('entryModulePath must be a string');
+  EventEmitter.call(this);
+  options = Object(options);
+  var keepAlive = options.keepAlive === undefined ? true : !!options.keepAlive;
```

Comment:
`.ref` and `.unref` on the object are different from keepAlive. An active worker keeps the parent alive, just like an active timer would, so there is ref and unref to change that behavior.

keepAlive option changes how the worker thread itself behaves when there is nothing on its event loop.